### PR TITLE
Support several warnings in tests

### LINKED ISSUE #4172: Add support to test for several warnings
> What is the problem you're trying to solve?

Currently it not possible to detect several warnings in tests, see
https://github.com/stylelint/stylelint/blob/master/jest-setup.js#L92

However, sometimes it is necessary for tests to detect such cases, see https://github.com/stylelint/stylelint/issues/4113

> What solution would you like to see?

My suggestion is to rework reject object syntax.

Typescript defintion:
```
type WarningDescription = {
    message: string,
    line?: number,
    column?: number,
    fixed?: string
};

type Reject = {code: string, warnings?: Array<WarningDescription>} & WarningDescription;
```

### PR DESCRIPTION
**Pull request by vankop**
> Which issue, if any, is this issue related to?

closes #4172 

> Is there anything in the PR that needs further explanation?

Nothing

### CODE REVIEW

File: jest-setup.js [hunk 1]
```diff
@@ -89,72 +89,91 @@ global.testRule = (rule, schema) => {
...
-
-                  if (testCase.message !== undefined) {
-                    expect(_.get(warning, "text")).toBe(testCase.message);
+                  if (
+                    schema.fix &&
+                    testCase.warnings &&
+                    testCase.warnings.length > 1
+                  ) {
+                    throw new Error(
+                      "Several expected warnings can not be used with { fix: true } in schema yet"
```
Comment (vankop) [inline jest-setup.js]: Don't know how to implement it better

Comment (hudochenkov) [reply jest-setup.js]: This is not needed. With `--fix` all warnings are fixed.

**Comment by vankop** [comment]
> I wonder why `.a, .a, .a {}` has only one warning then

it was fixed in https://github.com/stylelint/stylelint/issues/4113 , right now we have one warning for one selector

**Comment by vankop** [comment]
So,
```
.a, .b, .a, .b, .a, .b {}
```
Will produce 2 warnings for `a` and for `b`
```
.a, .a, .a {}
```
only for `.a`

File: jest-setup.js (hunk 1 above)
Comment (vankop) [reply jest-setup.js]: So you suggest to allow several `fixed` in `warnings` ?

Comment (hudochenkov) [reply jest-setup.js]: Based on how `fix` works we'll need only one `fixed` per case.

Comment (vankop) [reply jest-setup.js]: done

File: jest-setup.js [hunk 2]
```diff
@@ -89,26 +89,16 @@ global.testRule = (rule, schema) => {
                 };
 
                 return stylelint(options).then(output => {
-                  const warning = output.results[0].warnings[0];
+                  const actualWarnings = output.results[0].warnings;
 
-                  expect(output.results[0].parseErrors).toEqual([]);
```
Comment (hudochenkov) [inline jest-setup.js]: We need this line.

Comment (vankop) [reply jest-setup.js]: yep, my fault. fixed

**Pull request merged**